src/main/java/com/smartbear/readyapi/plugin/git/ui/AbstractRepositorySelectionGui.java

```diff
@@ -1,88 +1,118 @@
+    protected RepositoryForm selected;
 
     protected Component createGui(String helpUrl, String helpText, String remoteRepositoryUrl) {
-        JPanel panel = new JPanel(new MigLayout("wrap", "8[grow,fill]8", "8[][][][grow,fill][]8"));
 
-        boolean httpUrl = isHttpUrl(remoteRepositoryUrl);
-        sshRepositoryForm = new SshRepositoryForm(httpUrl ? null : remoteRepositoryUrl, false);
-        httpsRepositoryForm = new HttpsRepositoryForm(httpUrl ? remoteRepositoryUrl : null, false);
+        boolean isHttp = isHttpUrl(remoteRepositoryUrl);
 
-        ButtonGroup group = new ButtonGroup();
+        sshRepositoryForm = new SshRepositoryForm(isHttp ? null : remoteRepositoryUrl, false);
+        httpsRepositoryForm = new HttpsRepositoryForm(isHttp ? remoteRepositoryUrl : null, false);
+
+        List<RepositorySelection> repositorySelections = getRepositorySelections();
 
-        JRadioButton sshRadioButton = createRadioButton(LABEL_SSH, group);
-        panel.add(sshRadioButton);
-        JRadioButton httpsRadioButton = createRadioButton(LABEL_HTTPS, group);
-        panel.add(httpsRadioButton);
-        JRadioButton localRadioButton = createRadioButton(LABEL_LOCAL, group);
-        panel.add(localRadioButton);
+        JPanel panel = new JPanel(new MigLayout("wrap", "8[grow,fill]8", "8" + repeat("[]", repositorySelections.size()) + "[grow,fill][]8"));
+
+        ButtonGroup group = new ButtonGroup();
 
-        cards.add(sshRepositoryForm.getComponent(), LABEL_SSH);
-        cards.add(httpsRepositoryForm.getComponent(), LABEL_HTTPS);
-        cards.add(localRepositoryForm.getComponent(), LABEL_LOCAL);
+        for (RepositorySelection repositorySelection : repositorySelections) {
+            panel.add(repositorySelection.radioButton);
+            group.add(repositorySelection.radioButton);
+            cards.add(repositorySelection.repositoryForm.getComponent(), repositorySelection.label);
+        }
 
         panel.add(cards);
         panel.add(createLabelLink(helpUrl, helpText));
-        (httpUrl ? httpsRadioButton : sshRadioButton).setSelected(true);
-        selectCard(httpUrl ? LABEL_HTTPS : LABEL_SSH);
+
+        boolean shown = false;
+        for (RepositorySelection repositorySelection : repositorySelections) {
+
+            if ((LABEL_HTTPS.equals(repositorySelection.label) && isHttp)
+                    || (LABEL_SSH.equals(repositorySelection.label) && !isHttp)) {
+                repositorySelection.radioButton.setSelected(true);
+                selectCard(repositorySelection.label);
+                showCard(repositorySelection.label);
+                shown = true;
+                break;
+            }
+        }
+        if (!shown) {
+            repositorySelections.stream().findFirst().ifPresent(repositorySelection -> {
+                repositorySelection.radioButton.setSelected(true);
+                selectCard(repositorySelection.label);
+                showCard(repositorySelection.label);
```

Comment:
Move these calls to a separate method as they are also used in the cycle above. And you probably can use jsut repositorySelections.get(0).

Reply:
Extracted to the separate method

Reply:
This is a common method which uses a result of the `getRepositorySelections()`. But the mentioned method can be overridden and return empty list (hypothetically). So I use streams.

Reply:
Ok.